Question_Categorization/clustering_and_tagging_classes.py

```diff
@@ -159,12 +171,13 @@ def agglomerative_clustering(self, vectors):
         '''
         try:
             #Distance threshold: Distance up to which merging happens.
-            cluster = AgglomerativeClustering(n_clusters = None, affinity = 'euclidean', linkage = 'ward', distance_threshold=1.35, compute_full_tree=True)
+            cluster = AgglomerativeClustering(n_clusters = None, affinity = 'euclidean', linkage = 'ward', distance_threshold=DISTANCE_THRESHOLD, compute_full_tree=True)
             clusters = cluster.fit_predict(vectors)
             return clusters
         
         except Exception as e:
-            raise e
+            err = 'Error in Clustering.agglomerative_clustering: ' + str(e)
+            raise Exception(err)
     
 
 class Tagging: 
```

Comment:
With your updates, "n" in get_top_n_keywords is now a fraction of the number of candidate terms to use, right? Rather than the number of keywords to fetch? Let's update the documentation to match the change.